### PR TITLE
web: lower `.card` priority

### PR DESCRIPTION
**Pull request by valerybugakov**
## Context

- Fixes [the issue](https://sourcegraph.slack.com/archives/C03CSAER9LK/p1673011815990209) with inconsistent CSS order in the production bundle for the `Card` component.
- The underlying issue https://github.com/sourcegraph/sourcegraph/issues/42217

To address the root cause, we need to invest more time in restructuring Wildcard exports, as described in the linked issue.

> Restructure exports in packages with CSS modules to enable tree-shaking without relying on the `sideEffects` property.

## Test plan

1. `sg start web-standalone-prod`
2. Check that the toast is styled correctly.

## App preview:

- [Web](https://sg-web-vb-card-css-order.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (+0.01 kb) | 0.00% (+0.01 kb) | 0.00% (0.00 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/vb/card-css-order/compare-report.html;tab=live_object) for a full comparison between the commits d6d1af7 and 6bf0cf8 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/wildcard/src/components/Card/Card.module.scss [hunk 1]
```diff
@@ -1,4 +1,6 @@
-.card {
+// :where() is used to fix the CSS ordering issue with the Toast component
+// See https: //github.com/sourcegraph/sourcegraph/issues/42217
```
Comment (leonore) [inline client/wildcard/src/components/Card/Card.module.scss:2]:
```suggestion
// See https://github.com/sourcegraph/sourcegraph/issues/42217
```

**Review by leonore — COMMENTED** [review]
I have no context/experience on .card usage so don't want to blindly approve but thank you for taking a look 😄

**Review by philipp-spiess — APPROVED** [review]
TIL about `:where` and it always having a specificity of `0` 😱 

https://developer.mozilla.org/en-US/docs/Web/CSS/:where

**Comment by philipp-spiess** [comment]
@valerybugakov Chromatic shows a regression for border-radius on some components

**Comment by philipp-spiess** [comment]
Yeah it looks like some `<button>` styling takes presedence over the Card styling which is not what we want :/ 

Screenshot 2023-01-10 at 15 15 12

**Comment by philipp-spiess** [comment]
I’m fixing this by adding a `!important` for now. I know, I know but when a card is used you always want to use the border radius anyways 😅. Let's chat tomorrow for how this could be better fixed 🤔 We could also bump the specificity of the `.toast` class instead? 

For now I'll merge this when tests are green so we have unblocked `main`.

File: client/wildcard/src/components/Card/Card.module.scss [hunk 2]
```diff
@@ -16,7 +18,7 @@
     border-width: 1px;
     border-style: solid;
     border-color: var(--card-border-color);
-    border-radius: var(--card-border-radius);
+    border-radius: var(--card-border-radius) !important;
```
Comment (limitedmage) [inline client/wildcard/src/components/Card/Card.module.scss:21]: Why is `!important` required here?

Comment (limitedmage) [reply client/wildcard/src/components/Card/Card.module.scss:21]: Sounds good! Can we add a comment here explaining that? I don't think we should have undocumented `!important`s.

Comment (valerybugakov) [reply client/wildcard/src/components/Card/Card.module.scss:21]: Thanks, @philipp-spiess!

**Pull request merged**